File: src/Practice/p5.java
package Practice;

import java.text.DecimalFormat;
import java.util.Scanner;

public class p5 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);


        System.out.print("Введите кол-во строк в матрице: ");
        int n = scanner.nextInt();

        System.out.print("Введите кол-во столбцов в матрице: ");
        int m = scanner.nextInt();


        double[][] matrix = new double[n][m];

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                System.out.print("Введите элемент в строке " + (i + 1) + " столбце " + (j + 1) + ": ");
                matrix[i][j] = scanner.nextDouble();
            }
        }


        System.out.println("Матрица:");
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                String fmat = new DecimalFormat("#0.00").format(matrix[i][j]);
                System.out.print(fmat + "\t");
            }
            System.out.println();
        }
        double[] sums = new double[m / 2];
        for (int j = 0; j < m; j += 2) {
            double sum = 0;
            int count = 0;
            for (int i = 0; i < n; i++) {
                if (matrix[i][j] < 0) {
                    sum += matrix[i][j];
                    count++;
                }
            }
            if (count > 0) {
                sums[j / 2] = sum / 2;
            }
        }


        System.out.println("Половины сумм отрицательных элементов в четных столбцах:");
        for (int i = 0; i < m / 2; i++) {
            System.out.println(sums[i]);
        }
    }
}

File: Python/p5.py
import math

n = int(input("Введите кол-во строк в матрице: "))
m = int(input("Введите кол-во столбцов в матрице: "))

matrix = []
for i in range(n):
    row = []
    for j in range(m):
        element = float(input(f"Введите элемент в строке {i+1} столбце {j+1}: "))
        row.append(element)
    matrix.append(row)

print("Матрица:")
for row in matrix:
    for element in row:
        print("{:.2f}".format(element), end="\t")
    print()

sums = []
for j in range(0, m, 2):
    sum = 0
    count = 0
    for i in range(n):
        if matrix[i][j] < 0:
            sum += matrix[i][j]
            count += 1
    if count > 0:
        sums.append(sum / 2)

print("Половины сумм отрицательных элементов в четных столбцах:")
for s in sums:
    print("{:.2f}".format(s))
